avogadro/rendering/solid_first_fs.glsl

```diff
@@ -66,6 +68,47 @@ float lerp(float a, float b, float f)
     return a + f * (b - a);
 }
 
+float rand(vec2 co){
+  return fract(sin(dot(co.xy ,vec2(12.9898,78.233))) * 43758.5453);
+}
+
+float depthToZ(float depth) {
+  float eyeZ = ((height * 0.57735) /2.0);
+  float near = 5.0;
+  float far = 600.0;
+  float depthNormalized = 2.0 * depth - 1.0;
+  return 2.0 * near * far / (far + near - depthNormalized * (far - near));
+}
+
+float calcBlur(float z, float pixelScale) {
+  return clamp(abs(z - 39.0), 0.0, 0.5*pixelScale);
+}
+
+vec4 applyBlur(vec2 texCoord) {
+  float pixelScale = max(width, height);
+  float origZ = depthToZ(texture2D(inDepthTex, texCoord).x);
+  float blurAmt = calcBlur(origZ, pixelScale);
+  
+  float total = 1.0;
+  vec4 color = texture2D(inRGBTex, texCoord);
+  // number of samples can be optimized.
```

Comment:
Thank you for the review. I'll definitely make time to address the suggestions. However, regarding this issue, the loop index can't be compared with a non-constant expression. Therefore, adding `numSamples` to the for loop won't function since its value isn't constant. Could also leads to error. Are you sure about this one??

Reply:
You're right, and I appreciate your careful consideration. While GLSL 1.2 does support dynamic looping, the behavior can indeed vary based on the GPU and driver support. Sticking with your current approach to ensure compatibility and stability is wise. Thank you for pointing this out, @guthichs, and your reviews and help are greatly appreciated!